src/test/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLBuilderTest.java

```diff
@@ -0,0 +1,225 @@
+    stageToTableNamesMap = new HashMap<>();
+    stageToTableNamesMap.put("leftStage", "bqTable1");
+    stageToTableNamesMap.put("rightStage", "bqTable2");
+    project = "project";
+    dataset = "dataset";
+
+    helper = spy(new BigQuerySQLBuilder(builder, project, dataset, stageToTableNamesMap));
+  }
+
+  @Test
+  public void testGetFromClause() {
+
+    JoinStage stage1 = JoinStage.builder("stage1", null).build();
+    JoinStage stage2 = JoinStage.builder("stage2", null).build();
+    JoinStage stage3 = JoinStage.builder("stage3", null).build();
+    JoinStage stage4 = JoinStage.builder("stage4", null).build();
+
+    Map<String, JoinKey> stageNameToJoinKeyMap = new HashMap<>();
+    List<JoinStage> stages = Arrays.asList(stage1, stage2, stage3, stage4);
+
+    doReturn("table1").when(helper).getBQTableNameForStage("stage1");
+    doAnswer(invocationOnMock -> {
+      builder.append("JOIN table2 on table1.a = table2.b");
+      return null;
+    })
+      .when(helper)
+      .addJoinOnKeyOperation(eq(stage1), eq(stage2), eq(stageNameToJoinKeyMap));
+    doAnswer(invocationOnMock -> {
+      builder.append("JOIN table3 on table2.c = table3.d");
+      return null;
+    })
+      .when(helper)
+      .addJoinOnKeyOperation(eq(stage2), eq(stage3), eq(stageNameToJoinKeyMap));
+    doAnswer(invocationOnMock -> {
+      builder.append("JOIN table4 on table3.e = table4.f");
+      return null;
+    })
+      .when(helper)
+      .addJoinOnKeyOperation(eq(stage3), eq(stage4), eq(stageNameToJoinKeyMap));
+
+    helper.addFieldEqualityFromClause(stages, stageNameToJoinKeyMap);
+
+    Assert.assertEquals("table1"
+                          + " JOIN table2 on table1.a = table2.b"
+                          + " JOIN table3 on table2.c = table3.d"
+                          + " JOIN table4 on table3.e = table4.f",
+                        builder.toString());
+  }
+
+  @Test
+  public void testBuildJoinOperationInnerJoin() {
+    Map<String, JoinKey> stringJoinKeyMap = new HashMap<>();
+    configureJoinOperationTest(stringJoinKeyMap);
+
+    JoinStage leftStage = JoinStage.builder("leftStage", null).setRequired(true).build();
+    JoinStage rightStage = JoinStage.builder("rightStage", null).setRequired(true).build();
+
+    helper.addJoinOnKeyOperation(leftStage, rightStage, stringJoinKeyMap);
+    Assert.assertEquals("INNER JOIN `rightTable` ON some join condition",
+                        builder.toString());
+  }
+
+  @Test
+  public void testBuildJoinOperationLeftJoin() {
+    Map<String, JoinKey> stringJoinKeyMap = new HashMap<>();
+    configureJoinOperationTest(stringJoinKeyMap);
+
+    JoinStage leftStage = JoinStage.builder("leftStage", null).setRequired(true).build();
+    JoinStage rightStage = JoinStage.builder("rightStage", null).setRequired(false).build();
+
+    helper.addJoinOnKeyOperation(leftStage, rightStage, stringJoinKeyMap);
+    Assert.assertEquals("LEFT OUTER JOIN `rightTable` ON some join condition",
```

Comment:
So, why do we quote table names only in some cases?

Reply:
So each tests is working on a different part of the code. When fully executed, the result will be a fully quoted table name.

For the purposes of some of my test, I just want to ensure that the output string from a given method has the right spacing/separation when building SQL.